### PR TITLE
Extended Fab

### PR DESCRIPTION
**Pull request by jcytron**
Moved the Action Button to React Native Paper so we could turn existing FAB into an extended FAB.

### CODE REVIEW

**Comment by jcytron** [comment]
[Action button](https://trello.com/c/D333GmTN/2853-action-button)

File: src/ActionButton/ActionButton.js [hunk 1]
```diff
@@ -10,47 +16,53 @@ export default class WrappedActionButton extends Component {
...
+      borderRadius: 24,
+      alignItems: 'center',
+      justifyContent: 'center',
     }
 
-    let iconStyles = { color }
-
+    let labelStyles = {}
+    if (this.props.styles && this.props.styles.text) {
+      console.log(this.props.styles.text)
```
Comment (nucks) [inline src/ActionButton/ActionButton.js:34]:
```suggestion
```

File: src/ActionButton/ActionButton.js [hunk 2]
```diff
@@ -10,47 +16,53 @@ export default class WrappedActionButton extends Component {
     color: '#fff',
   }
 
-  renderSub() {
-    let { color, backgroundColor, icon, action } = this.props
+  renderPaper() {
+    let { text, icon, color, backgroundColor, action, _fonts } = this.props
+    console.log('props: ', this.props)
```
Comment (nucks) [inline src/ActionButton/ActionButton.js:21]:
```suggestion
```

File: src/ActionButton/ActionButton.js [hunk 3]
```diff
@@ -10,47 +16,51 @@ export default class WrappedActionButton extends Component {
     color: '#fff',
   }
 
-  renderSub() {
-    let { color, backgroundColor, icon, action } = this.props
+  renderPaper() {
```
Comment (scott-massey) [inline src/ActionButton/ActionButton.js:19]: We should delete `renderSub` if we're no longer using it. If it's already deleted, we should rename this to be `renderSub`

File: src/ActionButton/manifest.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
   "displayName": "Action Button",
   "icon": "icon.png",
-  "defaultWidth": 56,
-  "defaultHeight": 56,
+  "defaultWidth": 48,
```
Comment (scott-massey) [inline src/ActionButton/manifest.json:4]: Do we still want to change this?

**Review by scott-massey — COMMENTED** [review]
Just a couple comments

File: src/ActionButton/ActionButton.js [hunk 5]
```diff
@@ -10,47 +11,109 @@ export default class WrappedActionButton extends Component {
...
+  state = {
+    width: null,
+  }
+
+  handleLayout = ({
+    nativeEvent: {
+      layout: { width },
+    },
+  }) => {
+    if (!this.state.width !== width) {
```
Comment (vkkostin) [inline src/ActionButton/ActionButton.js:23]: Should this be `this.state.width !== width`?

File: src/ActionButton/ActionButton.js [hunk 6]
```diff
@@ -10,47 +11,109 @@ export default class WrappedActionButton extends Component {
...
+          icon={({ size, color }) => (
+            <Icon name={icon} style={{ color }} size={size}></Icon>
+          )}
           onPress={action}
-          style={{
-            container: containerStyles,
-            icon: iconStyles,
-          }}
-        />
+          label={breakless ? breakless : null}
```
Comment (vkkostin) [inline src/ActionButton/ActionButton.js:107]: Can this just be `label={breakless}`? It's already falling back to `null` in its definition.

File: src/ActionButton/ActionButton.js [hunk 7]
```diff
@@ -10,47 +11,109 @@ export default class WrappedActionButton extends Component {
...
+      labelStyles = {
+        ...this.props.styles.text,
+      }
+    } else if (_fonts) {
+      labelStyles = {
+        fontFamily: _fonts.body,
+      }
+    }
+
+    let breakless =
```
Comment (vkkostin) [inline src/ActionButton/ActionButton.js:92]: Can we turn `let`s to `const`s? I think `breakless`, `wrapperStyles`, `containerStyles`, and the destructured `prop`s can be `const` bindings, maybe other things as well.

**Review by vkkostin — CHANGES_REQUESTED** [review]

File: src/ActionButton/ActionButton.js (hunk 5 above)
Comment (scott-massey) [reply src/ActionButton/ActionButton.js:23]: Weird how that slipped in there. On it.

File: src/ActionButton/ActionButton.js (hunk 7 above)
Comment (scott-massey) [reply src/ActionButton/ActionButton.js:92]: Yeah I can clean this up

File: src/ActionButton/ActionButton.js (hunk 6 above)
Comment (scott-massey) [reply src/ActionButton/ActionButton.js:107]: Right. I'll clean that up.

**Review by vkkostin — APPROVED** [review]

**Pull request merged**